ci-operator/step-registry/ipi/conf/inframachineset/ipi-conf-inframachineset-commands.sh

```diff
@@ -0,0 +1,186 @@
+then
+	echo "No kubeconfig, can not continue."
+	exit 0
+fi
+
+# Download jq
+mkdir /tmp/bin
+export PATH=$PATH:/tmp/bin
+curl -sL https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 > /tmp/bin/jq
+chmod ug+x /tmp/bin/jq
+
+# Create infra machineconfigpool
+oc create -f - <<EOF
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: infra
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,infra]}
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/infra: ""
+EOF
+
+# Get machineset name to generate a generic template
+ref_machineset_name=$(oc -n openshift-machine-api get -o 'jsonpath={range .items[*]}{.metadata.name}{"\n"}{end}' machinesets | grep worker | head -n1)
+
+# Replace machine name worker to infra
+infra_machineset_name="${ref_machineset_name/worker/infra}"
+
+export ref_machineset_name infra_machineset_name
+# Get a templated json from worker machineset, change machine type and machine name
+# and pass it to oc to create a new machine set
+oc get machineset $ref_machineset_name -n openshift-machine-api -o json |
+  jq --arg infra_node_type "${INFRA_NODE_TYPE}" \
+     --arg infra_machineset_name "${infra_machineset_name}" \
+     '
+      .metadata.name = $infra_machineset_name |
+      .spec.selector.matchLabels."machine.openshift.io/cluster-api-machineset" = $infra_machineset_name |
+      .spec.template.spec.providerSpec.value.instanceType = $infra_node_type |
+      .spec.template.metadata.labels."machine.openshift.io/cluster-api-machineset" = $infra_machineset_name |
+      .spec.template.spec.metadata.labels."node-role.kubernetes.io/infra" = "" |
+      del(.status) |
+      del(.metadata.selfLink) |
+      del(.metadata.uid)
+     ' | oc create -f -
+
+# Scale machineset to expected number of replicas
+oc -n openshift-machine-api scale machineset/"${infra_machineset_name}" --replicas="${INFRA_NODE_REPLICAS}"
+
+# Wait for infra nodes to come up
+while [[ $(oc -n openshift-machine-api get machineset/${infra_machineset_name} -o 'jsonpath={.status.readyReplicas}') != "${INFRA_NODE_REPLICAS}" ]]; do echo -n "." && sleep 5; done
+
+# Collect infra node names
+mapfile -t INFRA_NODE_NAMES < <(echo "$(oc get nodes -l node-role.kubernetes.io/infra -o name)" | sed 's;node\/;;g')
+
+# Move router
+oc apply -f - <<EOF
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  nodePlacement:
+    nodeSelector:
+      matchLabels:
+        node-role.kubernetes.io/infra: ""
+EOF
+
+INGRESS_PODS_MOVED="false"
+for i in $(seq 0 60); do
+  echo "Checking ingress pods, attempt ${i}"
+  mapfile -t INGRESS_NODES < <(oc get pods -n openshift-ingress -o jsonpath='{.items[*].spec.nodeName}')
+  if [[ -z "$("${INGRESS_NODES[@]}" "${INFRA_NODE_NAMES[@]}" | tr ' ' '\n' | sort | uniq -u)" ]]; then
```

Comment:
Not clear to me where the line 84 vs. 85 discrepancy comes in, but the 4.9-to-4.10 job [has][1]:

```
INFO[2021-11-23T20:58:17Z] Step e2e-aws-upgrade-infra-ipi-conf-inframachineset succeeded after 4m50s.
```

[despite][2]:

```
machineset.machine.openshift.io/ci-op-2viiphim-1189a-lthwm-infra-us-east-1c scaled
.....................................Warning: resource ingresscontrollers/default is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
ingresscontroller.operator.openshift.io/default configured
Checking ingress pods, attempt 0
/bin/bash: line 85: ip-10-0-165-52.ec2.internal ip-10-0-165-61.ec2.internal: command not found
```

Are we missing an `echo` or `join` or some such here?

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/21175/rehearse-21175-periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-upgrade-infra/1463238862422675456#1:build-log.txt%3A29
[2]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/21175/rehearse-21175-periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-aws-upgrade-infra/1463238862422675456/artifacts/e2e-aws-upgrade-infra/ipi-conf-inframachineset/build-log.txt

Reply:
c5d9807914 -> 47f66a774f adds an `echo`, and we can mark this thread resolved.